invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/components/RecordDeletion/DeletionModal.js

```diff
@@ -0,0 +1,356 @@
+    this.setState({ checklistState: nextChecklistState, messages: newMessages });
+  };
+
+  handleCheckboxUpdate = (index) => {
+    const { checkboxState } = this.state;
+    const nextCheckboxState = checkboxState.map((c, i) => {
+      if (i === index) {
+        return !checkboxState[i];
+      } else {
+        return c;
+      }
+    });
+    this.setState({ checkboxState: nextCheckboxState });
+  };
+
+  deletionRequestSchema = (immediateDeletionAllowed) => {
+    if (immediateDeletionAllowed) {
+      return Yup.object({
+        reason: Yup.string().required("Required"),
+      });
+    } else {
+      return Yup.object({
+        reason: Yup.string().required("Required"),
+        comment: Yup.string()
+          .min(25, "Please write at least 25 characters")
+          .required("Required"),
+      });
+    }
+  };
+
+  handleSubmit = async (values) => {
+    this.setState({ loading: true });
+    const { record } = this.props;
+    const payload = {
+      reason: values.reason,
+      comment: values.comment,
+    };
+    if ("request_deletion" in record.links) {
+      this.cancellableAction = withCancel(
+        http.post(record.links.request_deletion, payload)
+      );
+      try {
+        const response = await this.cancellableAction.promise;
+        const data = response.data;
+
+        if (response.status === 200) {
+          window.location.reload();
+        } else if (response.status === 201) {
+          window.location.href = data.links.self_html;
+        }
+      } catch (error) {
+        this.setState({ error: error });
+        console.error(error);
+      } finally {
+        this.setState({ loading: false });
+      }
+    } else {
+      this.setState({ error: "Could not submit deletion request", loading: false });
+    }
+  };
+
+  render() {
+    const { open, recordDeletion, options } = this.props;
+    const { loading, error, checklistState, checkboxState, messages } = this.state;
+
+    const { checklist } = recordDeletion;
+
+    if (!("recordDeletion" in recordDeletion)) {
+      return null;
+    }
+
+    const immediateDeletionAllowed =
+      recordDeletion["recordDeletion"]["immediate_deletion"]["allowed"];
+
+    const modalHeaderText = immediateDeletionAllowed
+      ? i18next.t("Delete record")
+      : i18next.t("Request deletion");
+    const deletionButtonText = immediateDeletionAllowed
+      ? i18next.t("Delete immediately")
+      : i18next.t("Request deletion");
+    const deletionButtonClass = immediateDeletionAllowed ? "negative" : "primary";
+
+    const files = recordDeletion["context"]["files"];
+    const internalDoi = recordDeletion["context"]["internalDoi"];
+
+    const formDisabled =
+      checkboxState.some((v) => v === false) ||
+      checklistState.some((x) => x === true || x === undefined);
+
+    return (
+      <Modal
+        open={open}
+        closeIcon
+        onClose={this.handleClose}
+        role="dialog"
+        aria-modal="true"
+        tab-index="-1"
```

Comment:
```suggestion
        tabIndex="-1"
```

Reply:
also ariaModal?

Reply:
Found this In React docs: 
all DOM properties and attributes (including event handlers) should be camelCased. For example, the HTML attribute tabindex corresponds to the attribute tabIndex in React. The exception is aria-* and data-* attributes, which should be lowercased. For example, you can keep aria-label as aria-label.
see: 
https://legacy.reactjs.org/docs/dom-elements.html#:~:text=In%20React%2C%20all,aria%2Dlabel.